src/java.base/macosx/classes/sun/nio/fs/BsdFileAttributeViews.java

```diff
@@ -49,28 +49,71 @@ private static void setTimes(UnixPath path, FileTime lastModifiedTime,
         // permission check
         path.checkWrite();
 
-        int commonattr = 0;
-        long modValue = 0L;
-        if (lastModifiedTime != null) {
-            modValue = lastModifiedTime.to(TimeUnit.NANOSECONDS);
-            commonattr |= UnixConstants.ATTR_CMN_MODTIME;
-        }
-        long accValue = 0L;
-        if (lastAccessTime != null) {
-            accValue = lastAccessTime.to(TimeUnit.NANOSECONDS);
-            commonattr |= UnixConstants.ATTR_CMN_ACCTIME;
-        }
-        long createValue = 0L;
-        if (createTime != null) {
-            createValue = createTime.to(TimeUnit.NANOSECONDS);
-            commonattr |= UnixConstants.ATTR_CMN_CRTIME;
-        }
-
+        int fd = -1;
         try {
-            setattrlist(path, commonattr, modValue, accValue, createValue,
-                        followLinks ?  0 : UnixConstants.FSOPT_NOFOLLOW);
-        } catch (UnixException x) {
-            x.rethrowAsIOException(path);
+            try {
+                fd = path.openForAttributeAccess(followLinks);
+            } catch (UnixException x) {
+                x.rethrowAsIOException(path);
```

Comment:
The try-open-catch shouldn't be nested in the other try-catch. If the open fails then the fd isn't closed and there is no suppressed exception.

Reply:
Corrected in BsdFileAttributeViews-setTimes-8298187.